app/components/UI/Tokens/util/sortAssets.test.ts

```diff
@@ -0,0 +1,263 @@
+    const sortedById = sortAssets(mockAssets, {
+      key: 'name',
+      sortCallback: 'alphaNumeric',
+      order: 'asc',
+    });
+
+    expect(sortedById[0].name).toBe('Asset A');
+    expect(sortedById[sortedById.length - 1].name).toBe('Asset Z');
+  });
+
+  test('sorts by balance in ascending order (stringNumeric)', () => {
+    const sortedById = sortAssets(mockAssets, {
+      key: 'balance',
+      sortCallback: 'stringNumeric',
+      order: 'asc',
+    });
+
+    expect(sortedById[0].balance).toBe('400');
+    expect(sortedById[sortedById.length - 1].balance).toBe('600');
+  });
+
+  test('sorts by balance in ascending order (numeric)', () => {
+    const sortedById = sortAssets(mockAssets, {
+      key: 'balance',
+      sortCallback: 'numeric',
+      order: 'asc',
+    });
+
+    expect(sortedById[0].balance).toBe('400');
+    expect(sortedById[sortedById.length - 1].balance).toBe('600');
+  });
+
+  test('sorts by profile.id in ascending order', () => {
+    const sortedById = sortAssets(mockAssets, {
+      key: 'profile.id',
+      sortCallback: 'stringNumeric',
+      order: 'asc',
+    });
+
+    expect(sortedById[0].profile.id).toBe('1');
+    expect(sortedById[sortedById.length - 1].profile.id).toBe('4');
+  });
+
+  test('sorts by profile.id in descending order', () => {
+    const sortedById = sortAssets(mockAssets, {
+      key: 'profile.id',
+      sortCallback: 'stringNumeric',
+      order: 'dsc',
+    });
+
+    expect(sortedById[0].profile.id).toBe('4');
+    expect(sortedById[sortedById.length - 1].profile.id).toBe('1');
+  });
+
+  test('sorts by deeply nested profile.info.category in ascending order', () => {
+    const sortedByCategory = sortAssets(mockAssets, {
+      key: 'profile.info.category',
+      sortCallback: 'alphaNumeric',
+      order: 'asc',
+    });
+
+    // Expecting the assets with defined categories to be sorted first
+    expect(sortedByCategory[0].profile.info?.category).toBe('bronze');
+    expect(
+      sortedByCategory[sortedByCategory.length - 1].profile.info?.category,
+    ).toBe('silver');
+  });
+
+  test('sorts by createdAt (date) in ascending order', () => {
+    const sortedByDate = sortAssets(mockAssets, {
+      key: 'createdAt',
+      sortCallback: 'date',
+      order: 'asc',
+    });
+
+    expect(sortedByDate[0].createdAt).toEqual(new Date('2021-07-20'));
+    expect(sortedByDate[sortedByDate.length - 1].createdAt).toEqual(
+      new Date('2023-01-01'),
+    );
+  });
+
+  test('sorts by createdAt (date) in descending order', () => {
+    const sortedByDate = sortAssets(mockAssets, {
+      key: 'createdAt',
+      sortCallback: 'date',
+      order: 'dsc',
+    });
+
+    expect(sortedByDate[0].createdAt).toEqual(new Date('2023-01-01'));
+    expect(sortedByDate[sortedByDate.length - 1].createdAt).toEqual(
+      new Date('2021-07-20'),
+    );
+  });
+
+  test('handles undefined deeply nested value gracefully when sorting', () => {
```

Comment:
```suggestion
  test('handles undefined deeply nested value when sorting', () => {
```
My recommendation is to try to avoid weasel words like "successfully" "gracefully"... the test passing or not is enough to qualify the result